features/google/src/context/google-context.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect, useState } from "react";
+import { FeatureStorage } from "@polypoly-eu/feature-storage";
+import { useHistory, useLocation } from "react-router-dom";
+import { importData } from "../model/importer";
+
+export const GoogleContext = React.createContext();
+
+//used until real storage is loaded
+const fakeStorage = {
+    files: null,
+    refreshFiles: async () => null,
+    readFile: async () => null,
+    removeFile: async () => {},
+};
+
+class RefreshFilesError extends Error {
+    constructor(cause) {
+        super("Failed to refresh files");
+        this.name = "RefreshFilesError";
+        this.cause = cause;
+    }
+}
+
+class FileImportError extends Error {
+    constructor(cause) {
+        super("Failed to import file");
+        this.name = "FileImportError";
+        this.cause = cause;
+    }
+}
+
+class FileSelectionError extends Error {
+    constructor(cause) {
+        super("Failed to select file");
+        this.name = "FileSelectionError";
+        this.cause = cause;
+    }
+}
+
+function updatePodNavigation(pod, history, handleBack, location) {
+    pod.polyNav.actions = {
+        back: () => handleBack(),
+    };
+    history.length > 1 &&
+    location.pathname !== "/overview" &&
+    location.pathname !== "/import"
+        ? pod.polyNav.setActiveActions(["back"])
+        : pod.polyNav.setActiveActions([]);
+}
+
+export const GoogleContextProvider = ({ children }) => {
+    const [pod, setPod] = useState(null);
+    const [files, setFiles] = useState(null);
+    const [storage, setStorage] = useState(fakeStorage);
+    const [globalError, setGlobalError] = useState(null);
+    const [selectedFile, setSelectedFile] = useState(null);
+    const [googleAccount, setGoogleAccount] = useState(null);
+
+    const location = useLocation();
+    const initPod = async () => await window.pod;
+    const history = useHistory();
+
+    function handleBack() {
+        if (history.length > 1) {
+            history.goBack();
+        }
+    }
+
+    async function runWithLoadingScreen(task) {
+        setFiles(null);
+        await task();
+        refreshFiles();
+    }
+
+    const handleSelectFile = async () => {
+        const { polyNav } = pod;
+        runWithLoadingScreen(async function () {
+            try {
+                setSelectedFile(await polyNav.pickFile("application/zip"));
+            } catch (error) {
+                setGlobalError(new FileSelectionError(error));
```

Comment:
So you store error as global state... But then when this returns, wouldn't you have to reject the promise? The global error will give you information about what happened, but you need to know first if something happened.

Reply:
We just copied the erroring logic from fbImport here. Will add the globalError errorPopUp though!

(Could also improve the erroring logic, but would need to do it in both features to ensure consistency)

Reply:
Not for this PR, but copying code is an indication that we need some common logic here. Being something as entangled with code as error handling might be tricky, but at least some guidelines might help. A global error handling and reporting framework would also help.